compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/LightTreePositioningStrategies.kt

```diff
@@ -368,15 +369,25 @@ object LightTreePositioningStrategies {
         }
     }
 
-    val REFERENCE_BY_QUALIFIED: LightTreePositioningStrategy = object : LightTreePositioningStrategy() {
+    val REFERENCE_BY_QUALIFIED: LightTreePositioningStrategy = FindReferencePositioningStrategy(false)
+    val REFERENCED_NAME_BY_QUALIFIED: LightTreePositioningStrategy = FindReferencePositioningStrategy(true)
+
+    class FindReferencePositioningStrategy(val locateReferencedName: Boolean) : LightTreePositioningStrategy() {
         override fun mark(
             node: LighterASTNode,
             startOffset: Int,
             endOffset: Int,
             tree: FlyweightCapableTreeStructure<LighterASTNode>
         ): List<TextRange> {
             if (node.tokenType == KtNodeTypes.CALL_EXPRESSION || node.tokenType == KtNodeTypes.CONSTRUCTOR_DELEGATION_CALL) {
-                return markElement(tree.referenceExpression(node) ?: node, startOffset, endOffset, tree, node)
+                return markElement(tree.referenceExpression(node, locateReferencedName) ?: node, startOffset, endOffset, tree, node)
+            }
+            if (node.tokenType == KtNodeTypes.IS_EXPRESSION || node.tokenType == KtNodeTypes.BINARY_WITH_TYPE ||
+                node.tokenType == KtNodeTypes.BINARY_EXPRESSION || node.tokenType == KtNodeTypes.POSTFIX_EXPRESSION ||
+                node.tokenType == KtNodeTypes.PREFIX_EXPRESSION || node.tokenType == KtNodeTypes.BINARY_EXPRESSION ||
+                node.tokenType == KtNodeTypes.WHEN_CONDITION_IN_RANGE
```

Comment:
Some of these are `BINARY_EXPRESSIONS` in `idea.formatter`, but I don't think you can use that.

But, you can probably introduce a similar, private token set here and name it to indicate what those are; then just use `in` instead?